File: day13/ans.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        h = [-stone for stone in stones]
        heapq.heapify(h)
        
        while len(h) >  1:
            f = heapq.heappop(h)
            s = heapq.heappop(h)
            if(f != s):
                heapq.heappush(h, f-s)
        
        return -heapq.heappop(h) if len(h) > 0 else 0
            

File: day13/ans.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        std::priority_queue<int> pq  = {std::begin(stones), std::end(stones)};
        while(pq.size() > 1) {
            auto first = pq.top();
            pq.pop();
            
            auto second = pq.top();
            pq.pop();
            
            if(first != second) {
                pq.push(std::abs(first - second));
            }
        }
        
        if(pq.empty()) {
            return 0;
        }
        
        return pq.top();
    }
};
